stratum/hal/lib/barefoot/bfrt_table_manager.cc

```diff
@@ -48,6 +58,26 @@ ::util::Status BfrtTableManager::PushForwardingPipelineConfig(
       absl::make_unique<P4InfoManager>(p4_info);
   RETURN_IF_ERROR(p4_info_manager->InitializeAndVerify());
 
+  // pthread_t t = 0;
+  // if (t > 0 && pthread_join(register_reset_thread_, nullptr) != 0) {
+  //   RETURN_ERROR(ERR_INTERNAL) << "Failed to join thread " << t;
+  // }
+  if (register_reset_thread_ == 0 /*&& mode_ != OPERATION_MODE_SIM*/) {
+    RegisterClearThreadData* data = new RegisterClearThreadData(this);
+    for (const auto& reg : p4_info.registers()) {
+      data->registers.push_back(reg);
+    }
+    int ret = pthread_create(&register_reset_thread_, nullptr,
+                             &BfrtTableManager::RegisterClearThreadFunc, data);
+    if (ret != 0) {
+      RETURN_ERROR(ERR_INTERNAL) << "Failed to spawn register reset thread.";
+    }
+  } else {
+    LOG(FATAL)
+        << "Multiple pipeline pushes are not supported with register resets.";
```

Comment:
Fix this before we merge this.

Maybe use the timer daemon with a single event instead.

Reply:
I think that's the reason we need the lock when we push the pipeline